src/main/java/zd/zero/waifu/motivator/plugin/personality/Wendi.kt

```diff
@@ -72,14 +77,31 @@ object Wendi : Disposable {
 
             messageBusConnection.subscribe(MotivationEventListener.TOPIC, object : MotivationEventListener {
                 override fun onEventTrigger(motivationEvent: MotivationEvent) {
-                    consumeEvent(motivationEvent)
+                    when (motivationEvent.type) {
+                        MotivationEvents.IDLE ->
+                            idleEventDebouncer.debounceAndBuffer(motivationEvent) {
+                                consumeEvents(it)
+                            }
+                        else -> singleEventDebouncer.debounce {
+                            consumeEvent(motivationEvent)
+                        }
+                    }
                 }
             })
         }
     }
 
+    private fun consumeEvents(bufferedMotivationEvents: List<MotivationEvent>) {
+        val emotionalState = emotionCore.deriveMood(bufferedMotivationEvents.first())
```

Comment:
So I did not want multiple IDLE events to affect the Emotional state.

However, I did want to display the idle events for all of the projects. 
So I just added buffering to the the debounce. It waits until events stop coming in, then it invokes a method once with a list of motivation events.

Sorry if this is difficult concept to understand.